tests/long/truffleruby_comment_test.sh

```diff
@@ -17,6 +17,6 @@ rvm truffleruby-head do ruby -v # status=0; match=/truffleruby/
 rvm remove truffleruby-head # status=0
 
 ## Test that the right version is installed (#4633)
-rvm install truffleruby-1.0.0-rc13 # status=0; match!=/Already installed/
-rvm truffleruby-1.0.0-rc13 do ruby -v # status=0; match=/truffleruby 1.0.0-rc13/
-rvm remove truffleruby-1.0.0-rc13 # status=0
```

Comment:
Here's a snippet of the logs I can see locally:
```
Recompiling the OpenSSL C extension (libssl version is incompatible)
The clang, opt and llvm-link tools, part of LLVM, do not appear to be available in $PATH.
```

Reply:
Yeah updating the truffleruby version here seems good.
(I think it was working fine in CI, so it's probably expected some tests only pass on CI.)

Reply:
Ah it was failing in https://github.com/rvm/rvm/actions/runs/8845813076/job/24290469426 I see, maybe the side effect of `macos-latest` now using a newer macOS version, maybe we should pin to a given macos-NN version to avoid surprises like this.

Reply:
Yep, looks like it, that job ran on 14.4 and the last green job https://github.com/rvm/rvm/actions/runs/8713093296/job/23900815353#step:1:4 ran on 12.7.
I think we should pin to `macos-12` for now to fix the macos CI, and that should hopefully fix macos CI.
We can still update the TruffleRuby version tested it, that's fine.

Reply:
Unfortunately, I don't know what could have happened. My change exposed the problem elsewhere. If we want to solve it properly, we have to go through the full way.

Reply:
Looking at these two logs, it seems `macos-latest` used to be macos 12.7 arm64 and now resolved to macos 14 aarch64. Pretty weird stuff.